### PR TITLE
Replace pipeline CI with action CI

### LINKED ISSUE #201: [Feature]: Clarify the supported versions of python and add tests for each
### What does the feature solve?

In this project it is not clear what versions of python are required, supported, and tested against.  This makes it harder for consumers of this project to know their requirements are.  It also makes it harder for the developers of this project as they don't know what language features they are allowed to use.  

### Describe the solution

1. Add documentation for the supported versions of python and the rationale and process.  
2. Update documentation, Pypi metadata to be accurate.
3. Add PR gate tests for the supported versions of python
4. Add CI gates and status badges for each supported version

### Have you considered any alternatives?

_No response_

### Additional context

As pytools  usage expands it is critical to communicate these things as well as provide roadmap to indicate how these change as python releases new versions.

### LINKED ISSUE #215: [Feature]: edk2-pytool-library requires a CI test to validate the package against all  Python versions supported
### What does the feature solve?

This feature helps to validate the package against  all Python versions supported  and works as expected.
Without this check its possible that a given release might work with 3.x and breaks with 3.y.

### Describe the solution

Add CI support to validate package against all supported Python versions. 

### Have you considered any alternatives?

_No response_

### Additional context

_No response_

### LINKED ISSUE #218: Make documentation easier to access
The documentation should be accessible by web.

### PR DESCRIPTION
**Pull request by Javagedes**
Replaces azure pipelines based CI with github action based CI. 

Closes #218 
Closes #201
Closes #215

### CODE REVIEW

**Review by spbrogan — COMMENTED** [review]
main repo readme with badges needs to be updated.

**Comment by Javagedes** [comment]
> main repo readme with badges needs to be updated.

repo readme has been updated with badges

File: .github/workflows/VariableProducer.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      node-versions: ${{ steps.set-node-versions.outputs.VERSION }}
+
+    steps:
+    - name: Produce Python Version
+      id: set-python-versions
+      run: echo "VERSION=$pythonversions" >> $GITHUB_OUTPUT
+
+    - name: Produce Node Version
+      id: set-node-versions
+      run: echo "VERSION=$nodeversions" >> $GITHUB_OUTPUT
```
Comment (makubacki) [inline .github/workflows/VariableProducer.yml:37]: I suggest adding an empty line to the end of each file.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -14,3 +14,7 @@ updates:
     directory: "/" # Location of package manifests
     schedule:
       interval: "daily"
+  - package-ecosystem: "pip"
```
Comment (makubacki) [inline .github/dependabot.yml:17]: Probably worth adding github actions to the file now

**Review by makubacki — APPROVED** [review]

File: .github/dependabot.yml (hunk 2 above)
Comment (Javagedes) [reply .github/dependabot.yml:17]: Added github-actions as a package-ecosystem per https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot

File: .github/workflows/VariableProducer.yml (hunk 1 above)
Comment (Javagedes) [reply .github/workflows/VariableProducer.yml:37]: Added empty lines where they were not present.

**Pull request merged**